Rebuild administration pannel fails after migrate to 3.5.0

<!--
Hello 👋 Thank you for submitting an issue.

Before you start, please make sure your issue is understandable and reproducible.
To make your issue readable make sure you use valid Markdown syntax.

https://guides.github.com/features/mastering-markdown/

Please ensure you have also read and understand the contributing guide.

https://github.com/strapi/strapi/blob/master/CONTRIBUTING.md#reporting-an-issue
-->

## Bug report

### Describe the bug

After migrate from strapi 3.4.6 to 3.5.0  with [this guide](https://strapi.io/documentation/developer-docs/latest/migration-guide/) the administration panel rebuild fails.

### Steps to reproduce the behavior

1. migrate steps in [migration guide](https://strapi.io/documentation/developer-docs/latest/migration-guide/)
2. rebuild admin panel with `npm run build -- --clean`

### Code snippets

Terminal error output:
```
Building your admin UI with development configuration ...

● Webpack █████████████████████████ building (70%) 386/457 modules 71 active
 babel-loader › .cache/admin/src/containers/SettingsPage/utils/defaultRoutes.js

Error: Module not found: Error: Can't resolve '/usr/src/containers/App/utils/customRoutes' in '/usr/src/app/.cache/admin/src/containers/App/utils'
    at /usr/src/app/node_modules/strapi-admin/index.js:71:23
    at finalCallback (/usr/src/app/node_modules/webpack/lib/Compiler.js:257:39)
    at onCompiled (/usr/src/app/node_modules/webpack/lib/Compiler.js:265:20)
    at /usr/src/app/node_modules/webpack/lib/Compiler.js:670:21
    at eval (eval at create (/usr/src/app/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:28:1)
    at /usr/src/app/node_modules/webpack/lib/Compilation.js:1173:13
    at /usr/src/app/node_modules/webpack/lib/Compilation.js:1096:25
    at /usr/src/app/node_modules/webpack/lib/Compilation.js:1017:13
    at /usr/src/app/node_modules/neo-async/async.js:2830:7
    at /usr/src/app/node_modules/neo-async/async.js:2830:7
    at done (/usr/src/app/node_modules/neo-async/async.js:2863:11)
    at /usr/src/app/node_modules/neo-async/async.js:2818:7
    at /usr/src/app/node_modules/webpack/lib/Compilation.js:1017:13
    at /usr/src/app/node_modules/neo-async/async.js:2830:7
    at /usr/src/app/node_modules/neo-async/async.js:2830:7
    at done (/usr/src/app/node_modules/neo-async/async.js:2863:11)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! app@0.1.0 build: `strapi build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the app@0.1.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/node/.npm/_logs/2021-02-18T12_35_36_183Z-debug.log

```

### System

- Node.js version: 14
- NPM version: 6
- Strapi version: 3.5.0
- Database: MongoDB
- Operating system: Docker, Alpine-Linux

I'm having the same issue (and another one have reported it in slack as well https://strapi.slack.com/archives/C0BNGCDNH/p1613652596225700) 

It doesn't happen when I build it locally in production mode, but it does when circleci is building:
local:
```
 NODE_ENV=production yarn build --clean
yarn run v1.22.5
$ strapi build --clean
Building your admin UI with production configuration ...

✔ Webpack
  Compiled successfully in 6.74s

Done in 8.03s.
```
Circleci:
```
Step 11/12 : RUN yarn build
 ---> Running in 17c668f36de0
yarn run v1.22.5
$ strapi build
Building your admin UI with production configuration ...
ℹ Compiling Webpack
Error: Module not found: Error: Can't resolve '/ee/admin/containers/App/utils/customRoutes' in '/usr/src/app/.cache/admin/src/containers/App/utils'
    at /usr/src/app/node_modules/strapi-admin/index.js:71:23
    at finalCallback (/usr/src/app/node_modules/webpack/lib/Compiler.js:257:39)
    at onCompiled (/usr/src/app/node_modules/webpack/lib/Compiler.js:265:20)
    at /usr/src/app/node_modules/webpack/lib/Compiler.js:670:21
    at eval (eval at create (/usr/src/app/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:28:1)
    at /usr/src/app/node_modules/webpack/lib/Compilation.js:1173:13
    at /usr/src/app/node_modules/webpack/lib/Compilation.js:1096:25
    at /usr/src/app/node_modules/webpack/lib/Compilation.js:1017:13
    at /usr/src/app/node_modules/neo-async/async.js:2830:7
    at /usr/src/app/node_modules/neo-async/async.js:2830:7
    at done (/usr/src/app/node_modules/neo-async/async.js:2863:11)
    at /usr/src/app/node_modules/neo-async/async.js:2818:7
    at /usr/src/app/node_modules/webpack/lib/Compilation.js:1017:13
    at /usr/src/app/node_modules/neo-async/async.js:2830:7
    at /usr/src/app/node_modules/neo-async/async.js:2830:7
    at done (/usr/src/app/node_modules/neo-async/async.js:2863:11)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
The command '/bin/sh -c yarn build' returned a non-zero code: 1
```

I can confirm that src seems to be the issue. Moved my working local build and tried building and now it fails:
```
 /usr/src/strapi-test/3.5.0-test$ yarn build
yarn run v1.22.5
$ strapi build
Building your admin UI with development configuration ...

● Webpack █████████████████████████ building (39%) 244/292 modules 48 active
 node_modules/path-to-regexp/index.js

Error: Module not found: Error: Can't resolve '/usr/src/containers/App/utils/customRoutes' in '/usr/src/strapi-test/3.5.0-test/.cache/admin/src/containers/App/utils'
    at /usr/src/strapi-test/3.5.0-test/node_modules/strapi-admin/index.js:71:23
    at finalCallback (/usr/src/strapi-test/3.5.0-test/node_modules/webpack/lib/Compiler.js:257:39)
    at onCompiled (/usr/src/strapi-test/3.5.0-test/node_modules/webpack/lib/Compiler.js:265:20)
    at /usr/src/strapi-test/3.5.0-test/node_modules/webpack/lib/Compiler.js:670:21
    at eval (eval at create (/usr/src/strapi-test/3.5.0-test/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:26:1)
    at /usr/src/strapi-test/3.5.0-test/node_modules/webpack/lib/Compilation.js:1173:13
    at /usr/src/strapi-test/3.5.0-test/node_modules/webpack/lib/Compilation.js:1096:25
    at /usr/src/strapi-test/3.5.0-test/node_modules/webpack/lib/Compilation.js:1017:13
    at /usr/src/strapi-test/3.5.0-test/node_modules/neo-async/async.js:2830:7
    at /usr/src/strapi-test/3.5.0-test/node_modules/neo-async/async.js:2830:7
    at done (/usr/src/strapi-test/3.5.0-test/node_modules/neo-async/async.js:2863:11)
    at /usr/src/strapi-test/3.5.0-test/node_modules/neo-async/async.js:2818:7
    at /usr/src/strapi-test/3.5.0-test/node_modules/webpack/lib/Compilation.js:1017:13
    at /usr/src/strapi-test/3.5.0-test/node_modules/neo-async/async.js:2830:7
    at /usr/src/strapi-test/3.5.0-test/node_modules/neo-async/async.js:2830:7
    at done (/usr/src/strapi-test/3.5.0-test/node_modules/neo-async/async.js:2863:11)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
``` 

Node, yarn, node_modules are all the same, just the path containing src makes it break

I confirm, it hapens to me on a simple fresh install.
If your path have a 'src' folder in it, you will have this error.

> Error: Module not found: Error: Can't resolve 'C:\XXX\src/containers/App/utils/customRoutes' in 'C:\XXXX\src\my-project\.cache\admin\src\containers\App\utils'

--- a/packages/strapi-admin/webpack.config.js
+++ b/packages/strapi-admin/webpack.config.js
@@ -192,9 +192,10 @@ module.exports = ({
         ENABLED_EE_FEATURES: JSON.stringify(options.features),
       }),
       new webpack.NormalModuleReplacementPlugin(/ee_else_ce(\.*)/, function(resource) {
-        const splitPath = resource.context.split(`${path.sep}src${path.sep}`);
-
-        let wantedPath = path.join(splitPath[0], 'src');
+        let wantedPath = path.join(
+          resource.context.substr(0, resource.context.lastIndexOf(`${path.sep}src${path.sep}`)),
+          'src'
+        );
 
         if (useEE) {
           resource.request = resource.request.replace(

